libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/baseChart/PluggableBaseChart.tsx

```diff
@@ -160,6 +160,24 @@ export class PluggableBaseChart extends AbstractPluggableVisualization {
         this.customControlsProperties = customControlsProperties;
     }
 
+    public getExecution(
+        options: IVisProps,
+        insight: IInsightDefinition,
+        executionFactory: IExecutionFactory,
+    ) {
+        const { dateFormat, executionConfig } = options;
+        const supportedControls: IVisualizationProperties = this.getSupportedControls(insight, options);
```

Comment:
This is suspiscious: the `this.getSupportedControls` call expects the visualization class to be in a certain state (it for example reads the `this.visualizationProperties`, `this.featureFlags` etc.) so it expects the class to have been initializaed and updated. Hence not all the data comes to `getExecution` from its parameters and it is not pure because of it. I am not sure if purity of this function is necessary for your goals, but it seems a reasonable goal, right?

Or at least we should think about changing the name to something like `getCurrentExecution` and not pass anything to it - so that the function really "just" returns what it would use for its own execution at that point in time.

But maybe even better approach would be make this function static, so that it does not have the chance to access the internals and just depend on its arguments (being pure "by default").

Reply:
The same issue is present in GeoChart, PivotTable seems to be ok.

Reply:
Purity of `getExecution()` was not important for us as we only care about calling its wrapper from BaseVisualization. We basically tried to follow the same pattern as is in `update(...)` or `renderVisualization(...)`.

Reply:
the current plug viz design is unfortunate and leads to this; imho following the already set-up path is for the best atm as things here are quite brittle and may break in subtle ways. what is happening here is a split-method refactoring.. things improve code-wise but design still sucks (no offense) :)

Reply:
btw imma slightly more concerned about the base visualization impl that (i think) may lead to double executions because update does one and then getExecution() does another. that could warrant some changes (if feasible.. again hitting bad lifecycle setup that can mess things up)